common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -195,7 +194,7 @@ public static CalendarInterval fromDayTimeString(String s, String from, String t
     } else {
       try {
         int sign = m.group(1) != null && m.group(1).equals("-") ? -1 : 1;
-        long days = m.group(2) == null ? 0 : toLongWithRange("day", m.group(3),
+        int days = m.group(2) == null ? 0 : (int) toLongWithRange("day", m.group(3),
```

Comment:
Use `Math.toIntExact` and in other similar places

Reply:
can we use `Math.toIntExact` for only days field? as you know, there are long -> int in months filed as well, should we cover both of them?

Reply:
This is OK here, we call `toLongWithRange` with `Integer.MAX_VALUE` so it's safe to cast to int.